tests/tests/src/lampo_tests.rs

```diff
@@ -509,3 +518,235 @@ pub fn decode_offer() -> error::Result<()> {
+
+    let node_id = PublicKey::from_str(&node1_id).expect("Wrong node id");
+
+    let events = node1.lampod().events();
+    let _ = node1.fund_wallet(101)?;
+    wait!(|| {
+        let Ok(Event::OnChain(OnChainEvent::NewBestBlock((_, height)))) =
+            events.recv_timeout(Duration::from_millis(100))
+        else {
+            return Err(());
+        };
+        if height.to_consensus_u32() == 101 {
+            return Ok(());
+        }
+        Err(())
+    });
+
+    let _ = node3.fund_wallet(101)?;
+    let events3 = node3.lampod().events();
+    wait!(|| {
+        let Ok(Event::OnChain(OnChainEvent::NewBestBlock((_, height)))) =
+            events3.recv_timeout(Duration::from_millis(100))
+        else {
+            return Err(());
+        };
+        if height.to_consensus_u32() == 202 {
+            return Ok(());
+        }
+        Err(())
+    });
+
+    // The Provider needs to pay this invoice
+    let result = node2
+        .liquidity()
+        .create_jit_invoice(1000000, "A new desc".to_string())?;
+
+    log::info!("This is the invoice: {}", result.clone().to_string());
+
+    // Funding channel
+    let response: json::Value = node3
+        .lampod()
+        .call(
+            "fundchannel",
+            request::OpenChannel {
+                node_id: node1.info.node_id.clone(),
+                amount: 100000,
+                public: true,
+                port: Some(node1.port.clone()),
+                addr: Some("127.0.0.1".to_string()),
+            },
+        )
+        .unwrap();
+    let events3 = node3.lampod().events();
+    wait!(|| {
+        while let Ok(event) = events3.recv_timeout(Duration::from_millis(10)) {
+            node3.fund_wallet(6).unwrap();
+            if let Event::Lightning(LightningEvent::ChannelReady {
+                counterparty_node_id,
+                ..
+            }) = event
+            {
+                if counterparty_node_id.to_string() == node1.info.node_id {
+                    return Err(());
+                }
+                return Ok(());
+            };
+            // check if lampo see the channel
+            let channels: response::Channels =
+                node3.lampod().call("channels", json::json!({})).unwrap();
+            if channels.channels.is_empty() {
+                return Err(());
+            }
+
+            if !channels.channels.first().unwrap().ready {
+                return Err(());
+            }
+
+            let channels: response::Channels =
+                node1.lampod().call("channels", json::json!({})).unwrap();
+
+            if channels.channels.is_empty() {
+                return Err(());
+            }
+
+            if channels.channels.first().unwrap().ready {
+                return Ok(());
+            }
+        }
+        node3.fund_wallet(6).unwrap();
+        Err(())
+    });
+    // This would return a timeout as json_pay will continue to wait till it get an
```

Comment:
where is the code where you are calling `accept_inbound_channel_from_trusted_peer_0conf`?